test/LogTest.php

```diff
@@ -0,0 +1,222 @@
+ * the University Library of Hamburg University of Technology with funding from
+ * the German Research Foundation and the European Regional Development Fund.
+ *
+ * LICENCE
+ * OPUS is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    Test
+ * @package     OpusTest
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace OpusTest;
+
+use Opus\Log;
+
+class LogTest extends \PHPUnit_Framework_TestCase
+{
+
+    private $logFile;
+
+    public function tearDown()
+    {
+        if (is_resource($this->logFile)) {
+            fclose($this->logFile);
+        }
+
+        parent::tearDown();
+    }
+
+    public function testSetLevel()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setLevel(\Zend_Log::DEBUG);
+        $debugMessage = 'Debug level message from testSetLevel';
+        $opusLog->debug($debugMessage);
+        $content = $this->readLog();
+
+        $level = $opusLog->getLevel();
+
+        $this->assertContains($debugMessage, $content);
+        $this->assertEquals(\Zend_Log::DEBUG, $level);
+    }
+
+    public function testSetLevelForFilterDisabling()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setLevel(null);
+
+        $debugMessage = 'Debug level message from testSetLevelWithNullLevel';
+        $opusLog->debug($debugMessage);
+
+        $content = $this->readLog();
+
+        $this->assertContains($debugMessage, $content);
+    }
+
+    public function testSetLevelToNullHavingCustomLevels()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->addPriority('TEST', 8);
+        $opusLog->setLevel(null);
+        $opusLog->addPriority('TLEVEL', 9);
+
+        $testMessage = 'Test level message';
+        $opusLog->test($testMessage);
+
+        $tlevelMessage = 'Tlevel message';
+        $opusLog->tlevel($tlevelMessage);
+
+        $content = $this->readLog();
+
+        $this->assertContains($testMessage, $content);
+        $this->assertContains($tlevelMessage, $content);
+    }
+
+    public function testSetLevelNotEffectingOtherFilters()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $filter = new \Zend_Log_Filter_Priority(\Zend_Log::WARN);
```

Comment:
Okay, we add another filter. That will always apply, because it won't be changed by `setLevel`. So we need to have a check before that showing a message gets through and then a check afterwards showing that it isn't despite changing the level of our filter using `setLevel`.